libkbfs/file_data.go

```diff
@@ -901,15 +953,46 @@ func (fd *fileData) shiftBlocksToFillHole(
 			immedParent.pblock.IPtrs[currIndex],
 			newImmedParent.pblock.IPtrs[newCurrIndex]
 
-		// Cache the new immediate parent as dirty.
+		// Cache the new immediate parent as dirty.  Also cache the
+		// old immediate parent's right-most leaf child as dirty, to
+		// make sure this path is captured in
+		// getNextDirtyFileBlockAtOffset calls.  TODO: this is
```

Comment:
We have to mark the leaf as dirty, otherwise when `getNextDirtyFileBlockAtOffset` is called in `ready`, it will not return any of the dirty parents.  It works by checking the leaves only, and then returning the paths to those leaves.  If we don't mark the leaves, the parents won't be readied or synced to the server.